pagecarton/core/local_html/index.php

```diff
@@ -151,10 +151,21 @@ function xcopy($source, $dest, $permissions = 0755)
 		}
 		$pcBase = $newDir;
 
+		//the value of PC_BASE in pagecarton.json is the full path of the directory
+		// where the "core" folder resides
 		$pcConfig = json_decode( @file_get_contents( 'pagecarton.json' ), true );
-		if( ! empty( $pcConfig['PC_BASE'] ) && is_dir( $pcConfig['PC_BASE'] ) && is_writable( $pcConfig['PC_BASE'] ) )
+		//sometimes user eroneously set core folder that is not writeable.
+		//Hence, show error as appropriate
+		if( ! empty( $pcConfig['PC_BASE'] ) && is_dir( $pcConfig['PC_BASE'] ) )
 		{
-			$pcBase = $pcConfig['PC_BASE'];
+			if( is_writable( $pcConfig['PC_BASE'] ) ){
+				$pcBase = $pcConfig['PC_BASE'];
+			}else{
+				//Ayoola should help provide more context to this (i.e. some error handler or
+				//so), instead of just pumping error to user's face :)
+				//TODO: have a cutom error handler (hint: Php Monolog library)
+				exit("Sorry, the server could not write to the your custom core directory: <i>{$pcConfig['PC_BASE']}</i>. Please provide the server with necessary permissions to this folder.");

```

Comment:
This is just perfect. At least, till someone with a computer science college degree comes onboard. :)